### PR TITLE
Live Reloading the TimeLock Block, Part 2: TransactionManagers Plumbing

### PR DESCRIPTION
**Pull request by jeremyk-91**
- [x] smoketest on shopping product

**Goals (and why)**:
We want to be able to live reload the timelock block.
Regardless of whether we use remoting or not, we'll need fair chunks of this part of the infrastructure.

**As of this PR, we still do not handle the 0 node case correctly. That is likely to be Part 3.**

**Implementation Description (bullets)**:
- Add a timelock runtime config, which is part of the Atlas runtime config.
- We don't support live reloading from embedded or leader services into timelock. If the install config uses timelock, then we can look at the runtime config. If the runtime config has a server list we use it, otherwise we use the one from install.
- We still need to keep support for the old configs for back-compat, but I've marked them as deprecated.
- +550/-127 but a lot of it is dependency locks.

To come: Docs, ETE Tests (parts 3 and 4 probably)

**Concerns (what feedback would you like?)**:
- Is it okay to push ServerListConfig down into atlasdb-feign?
- AtlasDbFeignTargetFactory now has one constructor that looks quite different because of what needs to be live reloadable, yet I'd rather not break the APIs.

**Where should we start reviewing?**:
- Pick either AtlasDbFeignTargetFactory (for bottom-up traversal) or TransactionManagers (top-down traversal).

**Priority (whenever / two weeks / yesterday)**: two weeks?

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/2622)

### CODE REVIEW

File: atlasdb-config/src/main/java/com/palantir/atlasdb/config/TimeLockClientConfigs.java [hunk 1]
```diff
@@ -27,4 +32,5 @@ public static TimeLockClientConfig copyWithClient(TimeLockClientConfig config, S
                 .client(namespace)
                 .build();
     }
+
```
Comment (gmaretic) [inline atlasdb-config/src/main/java/com/palantir/atlasdb/config/TimeLockClientConfigs.java]: (nit)

File: atlasdb-config/src/main/java/com/palantir/atlasdb/config/AtlasDbRuntimeConfig.java [hunk 2]
```diff
@@ -58,6 +60,15 @@ public long getTimestampCacheSize() {
         return AtlasDbConstants.DEFAULT_TIMESTAMP_CACHE_SIZE;
     }
 
+    /**
+     * Runtime live-reloadable parameters for communicating with TimeLock.
+     *
+     * This value is ignored if the install config does not specify usage of TimeLock.
+     * We do not currently support live reloading from a leader block or using embedded services to using TimeLock.
+     */
+
```
Comment (gmaretic) [inline atlasdb-config/src/main/java/com/palantir/atlasdb/config/AtlasDbRuntimeConfig.java]: (nit)

Comment (jeremyk-91) [reply atlasdb-config/src/main/java/com/palantir/atlasdb/config/AtlasDbRuntimeConfig.java]: (removed)

File: atlasdb-config/src/main/java/com/palantir/atlasdb/config/TimeLockClientConfigs.java (hunk 1 above)
Comment (jeremyk-91) [reply atlasdb-config/src/main/java/com/palantir/atlasdb/config/TimeLockClientConfigs.java]: (removed)

File: atlasdb-feign/src/main/java/com/palantir/atlasdb/http/AtlasDbFeignTargetFactory.java [hunk 3]
```diff
@@ -118,6 +123,23 @@ private AtlasDbFeignTargetFactory() {
                 userAgent);
     }
 
+    public static <T> T createProxyWithFailover(
```
[the remark this answers is not in the document]
Comment (jeremyk-91) [reply atlasdb-feign/src/main/java/com/palantir/atlasdb/http/AtlasDbFeignTargetFactory.java]: I've renamed this to `createLiveReloadingProxyWithFailover` for now.

**Comment by jeremyk-91** [comment]
Need to switch to use new Refreshable

**Pull request merged**